cmd/snapd/main.go

```diff
@@ -56,6 +56,9 @@ func run() error {
 	t0 := time.Now().Truncate(time.Millisecond)
 	httputil.SetUserAgentFromVersion(cmd.Version)
 
+	ch := make(chan os.Signal)
+	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
reading here https://golang.org/pkg/os/signal/#Notify and considering that now there could be a span of time between getting a signal while in the initialisation code and reaching the select, we probably need ch to be buffered with at least buffer size 2